devlib/utils/ssh.py

```diff
@@ -321,6 +322,21 @@ def _scp(self, source, dest, timeout=30):
         except TimeoutError as e:
             raise TimeoutError(command_redacted, e.output)
 
+    def _sendline(self, command):
+        # Workaround for https://github.com/pexpect/pexpect/issues/552
+        if len(command) == self._get_window_size()[1] - self._get_prompt_length():
+            command += ' '
+        self.conn.sendline(command)
+
+    @memoized
+    def _get_prompt_length(self):
+        self.conn.sendline()
+        self.conn.prompt()
+        return len(self.conn.after)
+
+    @memoized
```

Comment:
This should not be memoized, as window size might change during the run  (e.g. terminal pane resized in tmux).

Reply:
Discussed offline and decided to leave as is to ensure this is as quick as possible and should not cause issues.